### DISCUSSION TITLE
Lampadina Zigbee LEDVANCE

### DISCUSSION THREAD

**Discussion by riccionese** [indomus/forum#1375] [Hardware & Firmware]
Evidentemente io ho dei problemi con Zigbee. Quantomeno con le lampadine, visto che tutti gli altri dispositivi e sensori vanno bene.

Non ho risolto i problemi con l'altra HUE anche seguendo i consigli ricevuti nell'altro mio post e allora ho comperata una nuova lampadina di altra marca per vedere se fosse un problema della HUE, del coordinator o ..."mio".
Ho Preso una LEDVANCE perché l'ho vista censita sul sito di Zigbee2Mqtt (https://www.zigbee2mqtt.io/devices/AC25702.html) e perché è un marchio che non ha un suo router/coordinator ma ne richiede uno di "terze parti".
Mi pareva una buona garanzia di integrabilità.

Fatto il pairing la lampadina si è registrata quasi regolarmente:
- la vedo accodata nel ```/opt/zigbee2mqtt/data/configuration.yaml``` (ovviamente il friendly_name l'ho cambiato io)

```
  '0x7cb03eaa0a07dbca':
    friendly_name: ZB-Bulb-1
```

- la vedo "available" anche in MQTT Explorer
![image](https://user-images.githubusercontent.com/62329292/150684580-6ea7e20b-b33c-4d26-bf7c-a5cd82e8468e.png)
- vedo anche nel pannello WEB di Ziigbee2Mqtt
![image](https://user-images.githubusercontent.com/62329292/150684671-9f41cdf0-1f09-46e9-82d5-5c0bc2436351.png)
... ma la vedo Disabilitata e Non supportata.
Ed Infatti anche nel TAB "espone" non c'è nulla di esposto.
Anche il log non dice niente di più
```
info  2022-01-23 14:46:21: ZB-Bulb-1 (0x7cb03eaa0a07dbca): Not supported (Unknown)
info  2022-01-23 14:46:25: MQTT publish: topic 'zigbee2mqtt/ZB-Bulb-1/availability', payload 'online'
error 2022-01-23 15:53:46: Failed to execute LQI for 'ZB-Bulb-1'
```

NOTA: vedo che OSRAM rimarca le LEDVANCE e la mia LEDVANCE ha come sigla AC03641 che sul sito di Zigbee2mqtt è riconosciuta come ORSAM, infatti --> https://www.zigbee2mqtt.io/devices/AC03641.html
Ma in ogni caso mi aspettavo fosse regolarmente supportata

qualcuno ha suggerimenti ?

**Comment by snakuzzo** [comment]
Sì...segui la procedura per farlo supportare...

https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html

**Reply by riccionese** [reply]
allora ... 
ho seguito il tuo consiglio (as usual) ed eseguendo passo passo la guida mi sono accorto che l'annuncio delle lampadina su MQTT avveniva (come dimostrato dallo screenshot postato sopra ed anche dal log) ma controllando MEGLIO il log ho trovato "pairing: failed"
ora: non riesco a capito come possa essere "failed" se il FE WEB di Z2M lo rileva e lo pubblica su MQTT ma ... boh !
Allora ho rimosso il device, ricontrollato il configuration.yaml, per sicurezza ho pure riavviato i servizi sia di HA sia di Z2M ed ho rieffettuato il pairing da ZERO,
Adesso ha rilevato tutto ed anche in "espone" vedo le funzioni disponibili ma ... non accetta comunque i comandi segnalando errori:
```
info  2022-01-24 14:40:23: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'state' to '0x7cb03eaa0a07dbca' failed: 'Error: Command 0x7cb03eaa0a07dbca/3 genOnOff.on({}, {\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'MAC no ack' (233))'","meta":{"friendly_name":"0x7cb03eaa0a07dbca"},"type":"zigbee_publish_error"}'
```

allora da vecchio informatico ho applicato la procedura standard: ho tolto corrente alla lampadina, ho contato fino a 10  e poi l'ho ricollegata
MIRACOLO! Ora vede tutto e comanda tutto (on/off e luminosità)

Come detto in altre occasioni... odio quando a parità di condizioni si ottengono risultati diversi ma oramai ho imparato a conviverci. 

Nota per i giovani che pensano l'informatica come una scienza esatta: in realtà si accorgeranno che è arte, casualità e magia ! 
;-)

**Reply by riccionese** [reply]
e comunque viene riconosciuta come OSRAM e non LEDVANCE

**Reply by snakuzzo** [reply]
Sul come viene riconosciuta non è un problema. E' del tutto normale...tanta roba di altre marche / sottomarche viene riconosciuta in maniera diversa...e tra le altre cose dipende anche da com'è stato implementato il tutto da zigbee2mqtt.  

Su questa affermazione...

**_Nota per i giovani che pensano l'informatica come una scienza esatta: in realtà si accorgeranno che è arte, casualità e magia !_**

...permettimi di dire che è una gran ca....ta. In informatica la casualità non esiste, e nemmeno la magia. Se succede qualcosa c'è sempre un motivo. Magari la tua affermazione può essere

**_Nota per i giovani...se in caso di difficoltà non sapete cosa fare e riavviando risolvete...sappiate che non è un caso...l'unica verità è che non siete stati in grado o non avete avuto l'occasione di capire qual era il problema_**

;)

**Reply by riccionese** [reply]
Me lo segno e ne riparliamo tra una 10ina di anni ! ;-DDD
